# HG changeset patch
# User Andrew John Hughes <ahughes@redhat.com>
# Date 1272471515 -3600
# Node ID c8d38c28900e01a04d730fc59f899ae0cd1d0779
# Parent  c30ad39537453fa171d094f3534e79a2faf16fad
PR476: Enable building SystemTap on GCC 4.5.

2010-04-28  Andrew John Hughes  <ahughes@redhat.com>

	PR icedtea/476
	* Makefile.am:
	Add patch when SystemTap is enabled to support
	building with GCC 4.5.
	* patches/systemtap-gcc-4.5.patch:
	Add cast to NULL (doesn't apply to DTrace due
	to differences between SystemTap and DTrace macros).

diff -r c30ad3953745 -r c8d38c28900e ChangeLog
--- a/ChangeLog	Mon Apr 26 14:11:34 2010 +0200
+++ b/ChangeLog	Wed Apr 28 17:18:35 2010 +0100
@@ -1,3 +1,13 @@
+2010-04-28  Andrew John Hughes  <ahughes@redhat.com>
+
+	PR icedtea/476
+	* Makefile.am:
+	Add patch when SystemTap is enabled to support
+	building with GCC 4.5.
+	* patches/systemtap-gcc-4.5.patch:
+	Add cast to NULL (doesn't apply to DTrace due
+	to differences between SystemTap and DTrace macros).
+
 2010-04-12  Andrew John Hughes  <ahughes@redhat.com>
 
 	PR icedtea/373
diff -r c30ad3953745 -r c8d38c28900e Makefile.am
--- a/Makefile.am	Mon Apr 26 14:11:34 2010 +0200
+++ b/Makefile.am	Wed Apr 28 17:18:35 2010 +0100
@@ -374,7 +374,8 @@
 endif
 
 if ENABLE_SYSTEMTAP
-ICEDTEA_PATCHES += patches/icedtea-systemtap.patch
+ICEDTEA_PATCHES += patches/icedtea-systemtap.patch \
+	patches/systemtap-gcc-4.5.patch
 endif
 
 if ENABLE_NSS
diff -r c30ad3953745 -r c8d38c28900e patches/systemtap-gcc-4.5.patch
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/patches/systemtap-gcc-4.5.patch	Wed Apr 28 17:18:35 2010 +0100
@@ -0,0 +1,11 @@
+--- openjdk.orig/hotspot/src/share/vm/prims/jni.cpp	2010-04-28 08:51:29.000000000 +0100
++++ openjdk/hotspot/src/share/vm/prims/jni.cpp	2010-04-28 09:29:22.000000000 +0100
+@@ -2723,7 +2723,7 @@
+ 
+   if (!directBufferSupportInitializeEnded) {
+     if (!initializeDirectBufferSupport(env, thread)) {
+-      DTRACE_PROBE1(hotspot_jni, NewDirectByteBuffer__return, NULL);
++      DTRACE_PROBE1(hotspot_jni, NewDirectByteBuffer__return, (uintptr_t) NULL);
+       return NULL;
+     }
+   }
